code/game/objects/effects/spawners/random/trash.dm

```diff
@@ -0,0 +1,237 @@
+/obj/effect/spawner/random/trash
+	name = "trash spawner"
+	desc = "Ewwwwwww gross."
+	icon_state = "trash"
+
+/obj/effect/spawner/random/trash/garbage
+	name = "garbage spawner"
+	loot = list(
+		/obj/effect/spawner/random/trash/food_packaging = 56,
+		/obj/item/trash/can = 8,
+		/obj/item/shard = 8,
+		/obj/effect/spawner/random/trash/botanical_waste = 8,
+		/obj/effect/spawner/random/trash/cigbutt = 8,
+		/obj/item/reagent_containers/syringe = 5,
+		/obj/item/light/tube/broken = 3,
+		/obj/item/food/deadmouse = 2,
+		/obj/item/light/tube/broken = 1,
+		/obj/item/trash/candle = 1,
+	)
+
+/obj/effect/spawner/random/trash/cigbutt
```

Comment:
throw the candy cigarette butt in here too

Reply:
Added with https://github.com/tgstation/tgstation/pull/60522/commits/b861b8d49e963ef85eee4af97e6086138a345698